{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cmb-panasas2/skchoudh/software_frozen/anaconda27/envs/riboraptor/lib/python3.6/site-packages/IPython/core/magics/pylab.py:160: UserWarning: pylab import has clobbered these variables: ['dedent']\n",
      "`%matplotlib` prevents importing * from pylab and numpy\n",
      "  \"\\n`%matplotlib` prevents importing * from pylab and numpy\"\n"
     ]
    }
   ],
   "source": [
    "%pylab inline\n",
    "import pandas as pd\n",
    "import os\n",
    "from tqdm import tqdm\n",
    "import fnmatch\n",
    "import errno    \n",
    "import glob\n",
    "from textwrap import dedent\n",
    "from riboraptor.helpers import mkdir_p\n",
    "from riboraptor.sradb import SRAdb\n",
    "from riboraptor.utils import copy_sra_data, create_config_file\n",
    "\n",
    "def mkdir_p(path):\n",
    "    \"\"\"Python version mkdir -p\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "\n",
    "    path : str\n",
    "    \"\"\"\n",
    "    if path:\n",
    "        try:\n",
    "            os.makedirs(path)\n",
    "        except OSError as exc:  # Python >2.5\n",
    "            if exc.errno == errno.EEXIST and os.path.isdir(path):\n",
    "                pass\n",
    "            else:\n",
    "                raise\n",
    "\n",
    "\n",
    "def symlink_force(source, destination):\n",
    "    \"\"\"Create forcelink forcefully\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    source: string\n",
    "            Location to source file\n",
    "    destination: string\n",
    "                 Location to target\n",
    "\n",
    "    \"\"\"\n",
    "    try:\n",
    "        os.symlink(source, destination)\n",
    "    except OSError as exc:\n",
    "        if exc.errno == errno.EEXIST:\n",
    "            os.remove(destination)\n",
    "            os.symlink(source, destination)\n",
    "        else:\n",
    "            raise exc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "re_ribo_root_dir = '/staging/as/skchoudh/SRA_datasets/'\n",
    "samples_to_process_dir = '/staging/as/skchoudh/re-ribo-datasets/'\n",
    "re_ribo_config_dir = '/home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs'\n",
    "re_ribo_analysis_dir = '/staging/as/skchoudh/re-ribo-analysis/'\n",
    "riboraptor_annotation_dir = '/home/cmb-panasas2/skchoudh/github_projects/riboraptor/riboraptor/annotation/'\n",
    "def mkdir_p(path):\n",
    "    \"\"\"Python version mkdir -p\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "\n",
    "    path : str\n",
    "    \"\"\"\n",
    "    if path:\n",
    "        try:\n",
    "            os.makedirs(path)\n",
    "        except OSError as exc:  # Python >2.5\n",
    "            if exc.errno == errno.EEXIST and os.path.isdir(path):\n",
    "                pass\n",
    "            else:\n",
    "                raise\n",
    "\n",
    "\n",
    "def symlink_force(source, destination):\n",
    "    \"\"\"Create forcelink forcefully\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    source: string\n",
    "            Location to source file\n",
    "    destination: string\n",
    "                 Location to target\n",
    "\n",
    "    \"\"\"\n",
    "    try:\n",
    "        os.symlink(source, destination)\n",
    "    except OSError as exc:\n",
    "        if exc.errno == errno.EEXIST:\n",
    "            os.remove(destination)\n",
    "            os.symlink(source, destination)\n",
    "        else:\n",
    "            raise exc\n",
    "genome_annotation_map = {\n",
    "    'hg38': 'v25',\n",
    "    'mm10': 'vM11',\n",
    "    'mg1655': '',\n",
    "    'sacCerR64': 'v91',\n",
    "    'MG1655': 'ASM584v2.38',\n",
    "    'BDGP6': 'v91',\n",
    "    'GRCz10': 'v91',\n",
    "    'panTro3': 'v94',\n",
    "    'Mmul8': 'v94'\n",
    "}\n",
    "\n",
    "\n",
    "genome_fasta_map = {\n",
    "    'hg38':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/hg38/fasta/hg38.fa',\n",
    "    'mm10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/mm10/fasta/mm10.fa',\n",
    "    'sacCerR64':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/sacCerR64/fasta/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa',\n",
    "    'MG1655':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/escherichia_coli_str_k_12_substr_mg1655/fasta/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.dna.toplevel.fa',\n",
    "    'BDGP6':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/drosophila_melanogaster_BDGP6/fasta/Drosophila_melanogaster.BDGP6.dna.toplevel.fa',\n",
    "    'GRCz10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/GRCz10/fasta/Danio_rerio.GRCz10.dna.toplevel.fa',\n",
    "    'panTro3':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/panTro3/fasta/Pan_troglodytes.Pan_tro_3.0.dna.toplevel.fa',\n",
    "    'Mmul8':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/Mmul8/fasta/Macaca_mulatta.Mmul_8.0.1.dna.toplevel.fa'\n",
    "}\n",
    "\n",
    "chrom_sizes_map = {\n",
    "    'hg38':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/hg38/fasta/hg38.chrom.sizes',\n",
    "    'mm10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/mm10/fasta/mm10.chrom.sizes',\n",
    "    'sacCerR64':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/sacCerR64/fasta/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.sizes',\n",
    "    'MG1655':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/escherichia_coli_str_k_12_substr_mg1655/fasta/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.dna.toplevel.sizes',\n",
    "    'BDGP6':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/drosophila_melanogaster_BDGP6/fasta/Drosophila_melanogaster.BDGP6.dna.toplevel.sizes',\n",
    "    'GRCz10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/GRCz10/fasta/Danio_rerio.GRCz10.dna.toplevel.sizes',\n",
    "    'panTro3':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/panTro3/fasta/Pan_troglodytes.Pan_tro_3.0.dna.toplevel.sizes',\n",
    "    'Mmul8':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/Mmul8/fasta/Macaca_mulatta.Mmul_8.0.1.dna.toplevel.sizes'\n",
    "}\n",
    "\n",
    "star_index_map = {\n",
    "    'hg38':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/hg38/star_annotated',\n",
    "    'mm10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/mm10/star_annotated',\n",
    "    'sacCerR64':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/sacCerR64/star_annotated',\n",
    "    'MG1655':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/escherichia_coli_str_k_12_substr_mg1655/star_annotated',\n",
    "    'BDGP6':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/drosophila_melanogaster_BDGP6/star_annotated',\n",
    "    'GRCz10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/GRCz10/star_annotated',\n",
    "    'panTro3':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/panTro3/star_annotated',\n",
    "    'Mmul8':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/Mmul8/star_annotated'\n",
    "}\n",
    "\n",
    "gtf_map = {\n",
    "    'hg38':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/hg38/annotation/gencode.v25.annotation.gtf',\n",
    "    'mm10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/mm10/annotation/gencode.vM11.annotation.gtf',\n",
    "    'sacCerR64':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/sacCerR64/annotation/Saccharomyces_cerevisiae.R64-1-1.91.gtf',\n",
    "    'MG1655':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/escherichia_coli_str_k_12_substr_mg1655/annotation/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.38.gtf',\n",
    "    'BDGP6':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/drosophila_melanogaster_BDGP6/annotation/Drosophila_melanogaster.BDGP6.91.gtf',\n",
    "    'GRCz10':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/GRCz10/annotation/Danio_rerio.GRCz10.91.gtf',\n",
    "    'Mmul8':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/Mmul8/annotation/Macaca_mulatta.Mmul_8.0.1.94.gtf',\n",
    "    'panTro3':\n",
    "    '/home/cmb-panasas2/skchoudh/genomes/panTro3/annotation/Pan_troglodytes.Pan_tro_3.0.94.gtf'\n",
    "}\n",
    "\n",
    "\n",
    "def filter_single_end_samples(df):\n",
    "    \"\"\"Filter single end samples from a dataframe\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    df: DataFrame\n",
    "        Dataframe as obtained from SRAb.sra_convert()\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    df: DataFrame\n",
    "        DataFrame with only single end samples\n",
    "    \"\"\"\n",
    "    df = df[~df['library_strategy'].str.contains('PAIRED')]\n",
    "    return df\n",
    "\n",
    "\n",
    "def copy_sra_data(df,\n",
    "                  sra_source_dir='/staging/as/skchoudh/SRA_datasets/',\n",
    "                  sra_dest_dir='/staging/as/skchoudh/re-ribo-datasets/'):\n",
    "    \"\"\"Copy SRA data to a new location retaining only single ended samples.\"\"\"\n",
    "    df = filter_single_end_samples(df)\n",
    "    assert len(df.study_accession.unique()) == 1, 'Multiple SRPs found'\n",
    "    srp = df.study_accession.unique()[0]\n",
    "    df_grouped = df.groupby(['taxon_id'])\n",
    "    srp_source_dir = os.path.join(sra_source_dir, srp)\n",
    "\n",
    "    for taxon_id, df_group in df_grouped:\n",
    "        species = taxon_id_map[taxon_id]\n",
    "        species_dest_dir = os.path.join(sra_dest_dir, species)\n",
    "        srp_dest_dir = os.path.join(species_dest_dir, srp)\n",
    "        mkdir_p(os.path.join(species_dest_dir, srp))\n",
    "        source_loc = srp_source_dir + os.path.sep + df_group[\n",
    "            'experiment_accession'].str.cat(\n",
    "                df_group['run_accession'] + '.sra', sep=os.path.sep)\n",
    "        dest_loc = srp_dest_dir + os.path.sep + df_group[\n",
    "            'experiment_accession'].str.cat(\n",
    "                df_group['run_accession'] + '.sra', sep=os.path.sep)\n",
    "        with tqdm(total=len(source_loc)) as pbar:\n",
    "            for source, dest in zip(source_loc, dest_loc):\n",
    "                mkdir_p(os.path.dirname(dest))\n",
    "                if os.path.isfile(source):\n",
    "                    symlink_force(source, dest)\n",
    "                pbar.update()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "db = SRAdb('/staging/as/skchoudh/SRAmetadb.sqlite')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "srp = 'SRP016501'\n",
    "df = db.sra_convert(srp)\n",
    "#copy_sra_data(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>experiment_title</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_name</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots</th>\n",
       "      <th>adapter_spec</th>\n",
       "      <th>avg_read_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081869</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>SRR306710</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>1480245008</td>\n",
       "      <td>19476908</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081870</td>\n",
       "      <td>GSM752558: gga br M 1</td>\n",
       "      <td>SRR306711</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752558: gga br M 1</td>\n",
       "      <td>1334334888</td>\n",
       "      <td>17557038</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081871</td>\n",
       "      <td>GSM752559: gga cb F 1</td>\n",
       "      <td>SRR306712</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752559: gga cb F 1</td>\n",
       "      <td>1795697524</td>\n",
       "      <td>23627599</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081872</td>\n",
       "      <td>GSM752560: gga cb M 1</td>\n",
       "      <td>SRR306713</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752560: gga cb M 1</td>\n",
       "      <td>1670739920</td>\n",
       "      <td>21983420</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081873</td>\n",
       "      <td>GSM752561: gga ht F 1</td>\n",
       "      <td>SRR306714</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752561: gga ht F 1</td>\n",
       "      <td>1748333260</td>\n",
       "      <td>23004385</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  study_accession experiment_accession       experiment_title run_accession  \\\n",
       "0       SRP007412            SRX081869  GSM752557: gga br F 1     SRR306710   \n",
       "1       SRP007412            SRX081870  GSM752558: gga br M 1     SRR306711   \n",
       "2       SRP007412            SRX081871  GSM752559: gga cb F 1     SRR306712   \n",
       "3       SRP007412            SRX081872  GSM752560: gga cb M 1     SRR306713   \n",
       "4       SRP007412            SRX081873  GSM752561: gga ht F 1     SRR306714   \n",
       "\n",
       "   taxon_id library_selection library_layout library_strategy  library_source  \\\n",
       "0      9031              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "1      9031              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "2      9031              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "3      9031              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "4      9031              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "\n",
       "            library_name       bases     spots adapter_spec  avg_read_length  \n",
       "0  GSM752557: gga br F 1  1480245008  19476908         None             76.0  \n",
       "1  GSM752558: gga br M 1  1334334888  17557038         None             76.0  \n",
       "2  GSM752559: gga cb F 1  1795697524  23627599         None             76.0  \n",
       "3  GSM752560: gga cb M 1  1670739920  21983420         None             76.0  \n",
       "4  GSM752561: gga ht F 1  1748333260  23004385         None             76.0  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "srp = 'SRP007412'\n",
    "df = db.sra_convert(srp)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 9031,  9258,  9544,  9593,  9597,  9598,  9600,  9606, 10090,\n",
       "       13616])"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df[df.taxon_id==9031]\n",
    "df.taxon_id.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxon_id_map = {\n",
    "    10090: 'mus_musculus',\n",
    "    9606: 'homo_sapiens',\n",
    "    4932: 'sacCerR64',\n",
    "    511145: 'MG1655',\n",
    "    7227: 'drosophila_melanogaster',\n",
    "    7955: 'danio_rerio',\n",
    "    9598: 'pan_troglodytes',\n",
    "    9544: 'macaca_mulatta',\n",
    "    9031: 'gallus_gallus',\n",
    "    9913: 'bos_taurus',\n",
    "    10116: 'rattus_norvegicus',\n",
    "    9258: 'ornithorhynchus_anatinus',\n",
    "    9593: 'gorilla_gorilla',\n",
    "    9597: 'pan_paniscus',\n",
    "    9600: 'pongo_abelii', #'pongo_pygmaeus',\n",
    "    13616: 'monodelphis_domestica'\n",
    "    \n",
    "}\n",
    "\n",
    "genomes_dir = '/home/cmb-panasas2/skchoudh/genomes'\n",
    "re_ribo_analysis_dir = '/staging/as/skchoudh/rna-seq-output'\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxon_id_map = {\n",
    "    10090: 'mus_musculus',\n",
    "    9606: 'homo_sapiens',\n",
    "    4932: 'sacCerR64',\n",
    "    511145: 'MG1655',\n",
    "    7227: 'drosophila_melanogaster',\n",
    "    7955: 'danio_rerio',\n",
    "    9598: 'pan_troglodytes',\n",
    "    9544: 'macaca_mulatta',\n",
    "    9031: 'gallus_gallus',\n",
    "    9913: 'bos_taurus',\n",
    "    10116: 'rattus_norvegicus',\n",
    "    9258: 'ornithorhynchus_anatinus',\n",
    "    9593: 'gorilla_gorilla',\n",
    "    9597: 'pan_paniscus',\n",
    "    9600: 'pongo_abelii', #'pongo_pygmaeus',\n",
    "    13616: 'monodelphis_domestica'\n",
    "    \n",
    "}\n",
    "\n",
    "genomes_dir = '/home/cmb-panasas2/skchoudh/genomes'\n",
    "\n",
    "cdna_map = {}\n",
    "for value in taxon_id_map.values():\n",
    "    genome_dir = os.path.join(genomes_dir, value)\n",
    "    if os.path.exists(os.path.join(genome_dir, 'cdna')):\n",
    "        cdna_map[value] = glob.glob('{}/*'.format(os.path.join(genome_dir, 'cdna')))[0]\n",
    "def create_config_file(df):\n",
    "    df_grouped = df.groupby(['taxon_id'])\n",
    "\n",
    "    for taxon_id, df_group in df_grouped:\n",
    "        assert len(\n",
    "            df_group['study_accession'].unique()) == 1, 'Multiple SRPs found'\n",
    "        species = taxon_id_map[taxon_id]\n",
    "        srp = df_group['study_accession'].unique()[0]\n",
    "        with open(\n",
    "                os.path.join(re_ribo_config_dir, '{}_{}.py'.format(\n",
    "                    species, srp)), 'w') as fh:\n",
    "\n",
    "            config = write_config(species, srp)\n",
    "            fh.write(config)\n",
    "            print('Wrote {}'.format(\n",
    "                os.path.join(re_ribo_config_dir, '{}_{}.py'.format(\n",
    "                    species, srp))))        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "def copy_sra_data(df,\n",
    "                  sra_source_dir='/staging/as/skchoudh/SRA_datasets/',\n",
    "                  sra_dest_dir='/staging/as/skchoudh/rna-seq-datasets/'):\n",
    "    \"\"\"Copy SRA data to a new location retaining only single ended samples.\"\"\"\n",
    "    assert len(df.study_accession.unique()) == 1, 'Multiple SRPs found'\n",
    "    srp = df.study_accession.unique()[0]\n",
    "    df_grouped = df.groupby(['taxon_id'])\n",
    "    srp_source_dir = os.path.join(sra_source_dir, srp)\n",
    "\n",
    "    for taxon_id, df_group in df_grouped:\n",
    "        species = taxon_id_map[taxon_id]\n",
    "        species_dest_dir = os.path.join(sra_dest_dir, species)\n",
    "        srp_dest_dir = os.path.join(species_dest_dir, srp)\n",
    "        mkdir_p(os.path.join(species_dest_dir, srp))\n",
    "        source_loc = srp_source_dir + os.path.sep + df_group[\n",
    "            'experiment_accession'].str.cat(\n",
    "                df_group['run_accession'] + '.sra', sep=os.path.sep)\n",
    "        dest_loc = srp_dest_dir + os.path.sep + df_group[\n",
    "            'experiment_accession'].str.cat(\n",
    "                df_group['run_accession'] + '.sra', sep=os.path.sep)\n",
    "        with tqdm(total=len(source_loc)) as pbar:\n",
    "            for source, dest in zip(source_loc, dest_loc):\n",
    "                mkdir_p(os.path.dirname(dest))\n",
    "                if os.path.isfile(source):\n",
    "                    symlink_force(source, dest)\n",
    "                pbar.update()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SRR_bamFile',\n",
       " 'SRX_bamFile',\n",
       " 'SRX_fastqFTP',\n",
       " 'adapter_spec',\n",
       " 'anonymized_name',\n",
       " 'base_caller',\n",
       " 'bases',\n",
       " 'center_project_name',\n",
       " 'common_name',\n",
       " 'description',\n",
       " 'design_description',\n",
       " 'experiment_ID',\n",
       " 'experiment_accession',\n",
       " 'experiment_alias',\n",
       " 'experiment_attribute',\n",
       " 'experiment_entrez_link',\n",
       " 'experiment_name',\n",
       " 'experiment_title',\n",
       " 'experiment_url_link',\n",
       " 'individual_name',\n",
       " 'instrument_model',\n",
       " 'instrument_name',\n",
       " 'library_construction_protocol',\n",
       " 'library_layout',\n",
       " 'library_name',\n",
       " 'library_selection',\n",
       " 'library_source',\n",
       " 'library_strategy',\n",
       " 'multiplier',\n",
       " 'number_of_levels',\n",
       " 'platform',\n",
       " 'platform_parameters',\n",
       " 'primary_study',\n",
       " 'qtype',\n",
       " 'quality_scorer',\n",
       " 'read_spec',\n",
       " 'related_studies',\n",
       " 'run_ID',\n",
       " 'run_accession',\n",
       " 'run_alias',\n",
       " 'run_attribute',\n",
       " 'run_center',\n",
       " 'run_date',\n",
       " 'run_entrez_link',\n",
       " 'run_url_link',\n",
       " 'sample_ID',\n",
       " 'sample_accession',\n",
       " 'sample_alias',\n",
       " 'sample_attribute',\n",
       " 'sample_entrez_link',\n",
       " 'sample_name',\n",
       " 'sample_url_link',\n",
       " 'sequence_space',\n",
       " 'spots',\n",
       " 'sradb_updated',\n",
       " 'study_ID',\n",
       " 'study_abstract',\n",
       " 'study_accession',\n",
       " 'study_alias',\n",
       " 'study_attribute',\n",
       " 'study_description',\n",
       " 'study_entrez_link',\n",
       " 'study_name',\n",
       " 'study_title',\n",
       " 'study_type',\n",
       " 'study_url_link',\n",
       " 'submission_ID',\n",
       " 'submission_accession',\n",
       " 'submission_center',\n",
       " 'submission_comment',\n",
       " 'submission_date',\n",
       " 'submission_lab',\n",
       " 'taxon_id',\n",
       " 'updated_date']"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(db.desc_table('sra_ft')['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 27/27 [00:00<00:00, 325.50it/s]\n",
      "100%|██████████| 27/27 [00:00<00:00, 369.01it/s]\n",
      "100%|██████████| 27/27 [00:00<00:00, 372.44it/s]\n",
      "100%|██████████| 26/26 [00:00<00:00, 393.26it/s]\n",
      "100%|██████████| 27/27 [00:00<00:00, 407.28it/s]\n"
     ]
    }
   ],
   "source": [
    "srp = 'SRP016501'\n",
    "df = db.sra_convert(srp)\n",
    "copy_sra_data(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP016501-gallus_gallus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP016501-macaca_mulatta.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP016501-bos_taurus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP016501-mus_musculus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP016501-rattus_norvegicus.tsv\n"
     ]
    }
   ],
   "source": [
    "srp = 'SRP016501'\n",
    "prefix = '/home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/{}'.format(srp)\n",
    "cols = ['study_accession',\n",
    " 'experiment_accession',\n",
    " 'experiment_title',\n",
    " 'run_accession',\n",
    " 'taxon_id',\n",
    " 'library_selection',\n",
    " 'library_layout',\n",
    " 'library_strategy',\n",
    " 'library_source',\n",
    " 'library_name',\n",
    " 'bases',\n",
    " 'spots',\n",
    " 'adapter_spec', 'description']\n",
    "df = db.sra_convert(srp, out_type=cols)\n",
    "df['tissue'] = df['experiment_title'].str.split(': ').str.get(1).str.split('; ').str.get(0)\n",
    "df = df.sort_values(by=['tissue', 'experiment_accession'])\n",
    "for taxon_id, group in df[['study_accession', 'experiment_accession', 'taxon_id', 'tissue']].drop_duplicates().groupby(['taxon_id']):\n",
    "    species = taxon_id_map[taxon_id]\n",
    "    filepath = '{}-{}.tsv'.format(prefix, species)\n",
    "    #rint(group)\n",
    "    group = group.sort_values(by=['tissue', 'experiment_accession'])\n",
    "    abundances = pd.DataFrame()\n",
    "    for index, row in group.iterrows():\n",
    "        srp =  row['study_accession']\n",
    "        tissue = ('_').join(row['tissue'].split('_')[1:])\n",
    "        srx = row['experiment_accession']\n",
    "        \n",
    "        abundance = pd.read_table('/staging/as/skchoudh/rna-seq-output/{}/{}/counts/{}/abundance.tsv'.format(species, srp, srx)) [['target_id', 'tpm']].rename(columns={'tpm': tissue}).set_index('target_id')\n",
    "        #    target_id       length  eff_length      est_counts      tpm\n",
    "\n",
    "        abundances = abundances.join(abundance, how='outer')\n",
    "    abundances = abundances.reset_index()\n",
    "    abundances.to_csv(filepath, header=True,  index=False, sep='\\t')\n",
    "    print('Wrote: {}'.format(filepath))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/gallus_gallus_SRP016501.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/macaca_mulatta_SRP016501.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/bos_taurus_SRP016501.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/mus_musculus_SRP016501.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/rattus_norvegicus_SRP016501.py\n"
     ]
    }
   ],
   "source": [
    "create_config_file(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-gallus_gallus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-ornithorhynchus_anatinus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-macaca_mulatta.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-gorilla_gorilla.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-pan_paniscus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-pan_troglodytes.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-pongo_abelii.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-homo_sapiens.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-mus_musculus.tsv\n",
      "Wrote: /home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/SRP007412-monodelphis_domestica.tsv\n"
     ]
    }
   ],
   "source": [
    "srp = 'SRP007412'\n",
    "df = db.sra_convert(srp)\n",
    "prefix = '/home/cmb-panasas2/skchoudh/github_projects/EE-546-project/cross-species-data/{}'.format(srp)\n",
    "\n",
    "#copy_sra_data(df)\n",
    "df = db.sra_convert(srp, out_type=cols+['experiment_name'])\n",
    "df['tissue'] = df['experiment_title'].str.split(': ').str.get(1).str[3:].str.lstrip(' ').str.replace(' ', '_')\n",
    "df = df.sort_values(by=['tissue', 'experiment_accession'])\n",
    "for taxon_id, group in df[['study_accession', 'experiment_accession', 'taxon_id', 'tissue']].drop_duplicates().groupby(['taxon_id']):\n",
    "    species = taxon_id_map[taxon_id]\n",
    "    filepath = '{}-{}.tsv'.format(prefix, species)\n",
    "    #rint(group)\n",
    "    group = group.sort_values(by=['tissue', 'experiment_accession'])\n",
    "    abundances = pd.DataFrame()\n",
    "    for index, row in group.iterrows():\n",
    "        srp =  row['study_accession']\n",
    "        tissue = row['tissue']\n",
    "        srx = row['experiment_accession']\n",
    "        \n",
    "        abundance = pd.read_table('/staging/as/skchoudh/rna-seq-output/{}/{}/counts/{}/abundance.tsv'.format(species, srp, srx)) [['target_id', 'tpm']].rename(columns={'tpm': tissue}).set_index('target_id')\n",
    "        #    target_id       length  eff_length      est_counts      tpm\n",
    "\n",
    "        abundances = abundances.join(abundance, how='outer')\n",
    "    abundances = abundances.reset_index()\n",
    "    abundances.to_csv(filepath, header=True,  index=False, sep='\\t')\n",
    "    print('Wrote: {}'.format(filepath))\n",
    "#create_config_file(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>experiment_title</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_name</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots</th>\n",
       "      <th>adapter_spec</th>\n",
       "      <th>description</th>\n",
       "      <th>experiment_name</th>\n",
       "      <th>avg_read_length</th>\n",
       "      <th>tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081869</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>SRR306710</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>1480245008</td>\n",
       "      <td>19476908</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>76.0</td>\n",
       "      <td>br_F_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081881</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>SRR306724</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>820587580</td>\n",
       "      <td>10797205</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>76.0</td>\n",
       "      <td>br_F_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081881</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>SRR306725</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>1850093840</td>\n",
       "      <td>24343340</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>76.0</td>\n",
       "      <td>br_F_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081893</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>SRR306742</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>860751832</td>\n",
       "      <td>11325682</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>76.0</td>\n",
       "      <td>br_F_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081893</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>SRR306743</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>3615666256</td>\n",
       "      <td>47574556</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>76.0</td>\n",
       "      <td>br_F_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_accession experiment_accession       experiment_title run_accession  \\\n",
       "0         SRP007412            SRX081869  GSM752557: gga br F 1     SRR306710   \n",
       "14        SRP007412            SRX081881  GSM752569: oan br F 1     SRR306724   \n",
       "15        SRP007412            SRX081881  GSM752569: oan br F 1     SRR306725   \n",
       "134       SRP007412            SRX081893  GSM752588: mdo br F 1     SRR306742   \n",
       "135       SRP007412            SRX081893  GSM752588: mdo br F 1     SRR306743   \n",
       "\n",
       "     taxon_id library_selection library_layout library_strategy  \\\n",
       "0        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "14       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "15       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "134     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "135     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "\n",
       "     library_source           library_name       bases     spots adapter_spec  \\\n",
       "0    TRANSCRIPTOMIC  GSM752557: gga br F 1  1480245008  19476908         None   \n",
       "14   TRANSCRIPTOMIC  GSM752569: oan br F 1   820587580  10797205         None   \n",
       "15   TRANSCRIPTOMIC  GSM752569: oan br F 1  1850093840  24343340         None   \n",
       "134  TRANSCRIPTOMIC  GSM752588: mdo br F 1   860751832  11325682         None   \n",
       "135  TRANSCRIPTOMIC  GSM752588: mdo br F 1  3615666256  47574556         None   \n",
       "\n",
       "    description        experiment_name  avg_read_length  tissue  \n",
       "0          None  GSM752557: gga br F 1             76.0  br_F_1  \n",
       "14         None  GSM752569: oan br F 1             76.0  br_F_1  \n",
       "15         None  GSM752569: oan br F 1             76.0  br_F_1  \n",
       "134        None  GSM752588: mdo br F 1             76.0  br_F_1  \n",
       "135        None  GSM752588: mdo br F 1             76.0  br_F_1  "
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>experiment_title</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_name</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots</th>\n",
       "      <th>adapter_spec</th>\n",
       "      <th>description</th>\n",
       "      <th>avg_read_length</th>\n",
       "      <th>tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [study_accession, experiment_accession, experiment_title, run_accession, taxon_id, library_selection, library_layout, library_strategy, library_source, library_name, bases, spots, adapter_spec, description, avg_read_length, tissue]\n",
       "Index: []"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.library_strategy.str.contains('PAIRED')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_paired = df[df.avg_read_length>76]\n",
    "df_unpaired = df[df.library_layout.str.contains('SINGLE')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>experiment_title</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_name</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots</th>\n",
       "      <th>adapter_spec</th>\n",
       "      <th>avg_read_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081869</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>SRR306710</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752557: gga br F 1</td>\n",
       "      <td>1480245008</td>\n",
       "      <td>19476908</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081870</td>\n",
       "      <td>GSM752558: gga br M 1</td>\n",
       "      <td>SRR306711</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752558: gga br M 1</td>\n",
       "      <td>1334334888</td>\n",
       "      <td>17557038</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081871</td>\n",
       "      <td>GSM752559: gga cb F 1</td>\n",
       "      <td>SRR306712</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752559: gga cb F 1</td>\n",
       "      <td>1795697524</td>\n",
       "      <td>23627599</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081872</td>\n",
       "      <td>GSM752560: gga cb M 1</td>\n",
       "      <td>SRR306713</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752560: gga cb M 1</td>\n",
       "      <td>1670739920</td>\n",
       "      <td>21983420</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081873</td>\n",
       "      <td>GSM752561: gga ht F 1</td>\n",
       "      <td>SRR306714</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752561: gga ht F 1</td>\n",
       "      <td>1748333260</td>\n",
       "      <td>23004385</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081874</td>\n",
       "      <td>GSM752562: gga ht M 1</td>\n",
       "      <td>SRR306715</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752562: gga ht M 1</td>\n",
       "      <td>1604929848</td>\n",
       "      <td>21117498</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081875</td>\n",
       "      <td>GSM752563: gga kd F 1</td>\n",
       "      <td>SRR306716</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752563: gga kd F 1</td>\n",
       "      <td>1749607628</td>\n",
       "      <td>23021153</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081876</td>\n",
       "      <td>GSM752564: gga kd M 1</td>\n",
       "      <td>SRR306717</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752564: gga kd M 1</td>\n",
       "      <td>1732548288</td>\n",
       "      <td>22796688</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081877</td>\n",
       "      <td>GSM752565: gga lv F 1</td>\n",
       "      <td>SRR306718</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752565: gga lv F 1</td>\n",
       "      <td>2298690376</td>\n",
       "      <td>30245926</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081878</td>\n",
       "      <td>GSM752566: gga lv M 1</td>\n",
       "      <td>SRR306719</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752566: gga lv M 1</td>\n",
       "      <td>653253896</td>\n",
       "      <td>8595446</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081878</td>\n",
       "      <td>GSM752566: gga lv M 1</td>\n",
       "      <td>SRR306720</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752566: gga lv M 1</td>\n",
       "      <td>1713238740</td>\n",
       "      <td>22542615</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081879</td>\n",
       "      <td>GSM752567: gga ts M 1</td>\n",
       "      <td>SRR306721</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752567: gga ts M 1</td>\n",
       "      <td>666191072</td>\n",
       "      <td>8765672</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081879</td>\n",
       "      <td>GSM752567: gga ts M 1</td>\n",
       "      <td>SRR306722</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752567: gga ts M 1</td>\n",
       "      <td>1632872768</td>\n",
       "      <td>21485168</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081880</td>\n",
       "      <td>GSM752568: gga ts M 2</td>\n",
       "      <td>SRR306723</td>\n",
       "      <td>9031</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752568: gga ts M 2</td>\n",
       "      <td>2804356756</td>\n",
       "      <td>36899431</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081881</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>SRR306724</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>820587580</td>\n",
       "      <td>10797205</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081881</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>SRR306725</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752569: oan br F 1</td>\n",
       "      <td>1850093840</td>\n",
       "      <td>24343340</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081882</td>\n",
       "      <td>GSM752570: oan br M 1</td>\n",
       "      <td>SRR306726</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752570: oan br M 1</td>\n",
       "      <td>707293012</td>\n",
       "      <td>9306487</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081882</td>\n",
       "      <td>GSM752570: oan br M 1</td>\n",
       "      <td>SRR306727</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752570: oan br M 1</td>\n",
       "      <td>1037813896</td>\n",
       "      <td>13655446</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081883</td>\n",
       "      <td>GSM752571: oan cb F 1</td>\n",
       "      <td>SRR306728</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752571: oan cb F 1</td>\n",
       "      <td>2302429576</td>\n",
       "      <td>30295126</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081884</td>\n",
       "      <td>GSM752572: oan cb M 1</td>\n",
       "      <td>SRR306729</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752572: oan cb M 1</td>\n",
       "      <td>2159580280</td>\n",
       "      <td>28415530</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081885</td>\n",
       "      <td>GSM752573: oan ht F 1</td>\n",
       "      <td>SRR306730</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752573: oan ht F 1</td>\n",
       "      <td>1429341652</td>\n",
       "      <td>18807127</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081886</td>\n",
       "      <td>GSM752575: oan ht M 1</td>\n",
       "      <td>SRR306731</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752575: oan ht M 1</td>\n",
       "      <td>1282580180</td>\n",
       "      <td>16876055</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081887</td>\n",
       "      <td>GSM752576: oan kd F 1</td>\n",
       "      <td>SRR306732</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752576: oan kd F 1</td>\n",
       "      <td>1357606316</td>\n",
       "      <td>17863241</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081888</td>\n",
       "      <td>GSM752578: oan kd M 1</td>\n",
       "      <td>SRR306733</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752578: oan kd M 1</td>\n",
       "      <td>883208008</td>\n",
       "      <td>11621158</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081888</td>\n",
       "      <td>GSM752578: oan kd M 1</td>\n",
       "      <td>SRR306734</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752578: oan kd M 1</td>\n",
       "      <td>1222176520</td>\n",
       "      <td>16081270</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081889</td>\n",
       "      <td>GSM752580: oan lv F 1</td>\n",
       "      <td>SRR306735</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752580: oan lv F 1</td>\n",
       "      <td>1205382724</td>\n",
       "      <td>15860299</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081889</td>\n",
       "      <td>GSM752580: oan lv F 1</td>\n",
       "      <td>SRR306736</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752580: oan lv F 1</td>\n",
       "      <td>1635037248</td>\n",
       "      <td>21513648</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081890</td>\n",
       "      <td>GSM752581: oan lv M 1</td>\n",
       "      <td>SRR306737</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752581: oan lv M 1</td>\n",
       "      <td>1377145688</td>\n",
       "      <td>18120338</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081890</td>\n",
       "      <td>GSM752581: oan lv M 1</td>\n",
       "      <td>SRR306738</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752581: oan lv M 1</td>\n",
       "      <td>1501827716</td>\n",
       "      <td>19760891</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081891</td>\n",
       "      <td>GSM752583: oan ts M 1</td>\n",
       "      <td>SRR306739</td>\n",
       "      <td>9258</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752583: oan ts M 1</td>\n",
       "      <td>1436304772</td>\n",
       "      <td>18898747</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081907</td>\n",
       "      <td>GSM752616: mmu br M 2</td>\n",
       "      <td>SRR306762</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752616: mmu br M 2</td>\n",
       "      <td>1499177976</td>\n",
       "      <td>19726026</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081908</td>\n",
       "      <td>GSM752617: mmu cb F 1</td>\n",
       "      <td>SRR306763</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752617: mmu cb F 1</td>\n",
       "      <td>3141899660</td>\n",
       "      <td>41340785</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081909</td>\n",
       "      <td>GSM752618: mmu cb M 1</td>\n",
       "      <td>SRR306764</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752618: mmu cb M 1</td>\n",
       "      <td>1592028164</td>\n",
       "      <td>20947739</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081910</td>\n",
       "      <td>GSM752619: mmu cb M 2</td>\n",
       "      <td>SRR306765</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752619: mmu cb M 2</td>\n",
       "      <td>2181979000</td>\n",
       "      <td>28710250</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081911</td>\n",
       "      <td>GSM752620: mmu ht F 1</td>\n",
       "      <td>SRR306766</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752620: mmu ht F 1</td>\n",
       "      <td>3394842784</td>\n",
       "      <td>44668984</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081912</td>\n",
       "      <td>GSM752621: mmu ht M 1</td>\n",
       "      <td>SRR306767</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752621: mmu ht M 1</td>\n",
       "      <td>1861519756</td>\n",
       "      <td>24493681</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081913</td>\n",
       "      <td>GSM752622: mmu ht M 2</td>\n",
       "      <td>SRR306768</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752622: mmu ht M 2</td>\n",
       "      <td>1968701036</td>\n",
       "      <td>25903961</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081914</td>\n",
       "      <td>GSM752623: mmu kd F 1</td>\n",
       "      <td>SRR306769</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752623: mmu kd F 1</td>\n",
       "      <td>3466613232</td>\n",
       "      <td>45613332</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081915</td>\n",
       "      <td>GSM752624: mmu kd M 1</td>\n",
       "      <td>SRR306770</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752624: mmu kd M 1</td>\n",
       "      <td>1796622064</td>\n",
       "      <td>23639764</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081916</td>\n",
       "      <td>GSM752625: mmu kd M 2</td>\n",
       "      <td>SRR306771</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752625: mmu kd M 2</td>\n",
       "      <td>2216025784</td>\n",
       "      <td>29158234</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081917</td>\n",
       "      <td>GSM752626: mmu lv F 1</td>\n",
       "      <td>SRR306772</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752626: mmu lv F 1</td>\n",
       "      <td>3671311252</td>\n",
       "      <td>48306727</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081918</td>\n",
       "      <td>GSM752627: mmu lv M 1</td>\n",
       "      <td>SRR306773</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752627: mmu lv M 1</td>\n",
       "      <td>1401775616</td>\n",
       "      <td>18444416</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081919</td>\n",
       "      <td>GSM752628: mmu lv M 2</td>\n",
       "      <td>SRR306774</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752628: mmu lv M 2</td>\n",
       "      <td>2584775808</td>\n",
       "      <td>34010208</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081920</td>\n",
       "      <td>GSM752629: mmu ts M 1</td>\n",
       "      <td>SRR306775</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752629: mmu ts M 1</td>\n",
       "      <td>1518001808</td>\n",
       "      <td>19973708</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081921</td>\n",
       "      <td>GSM752630: mmu ts M 2</td>\n",
       "      <td>SRR306776</td>\n",
       "      <td>10090</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752630: mmu ts M 2</td>\n",
       "      <td>2261985720</td>\n",
       "      <td>29762970</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081893</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>SRR306742</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>860751832</td>\n",
       "      <td>11325682</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081893</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>SRR306743</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752588: mdo br F 1</td>\n",
       "      <td>3615666256</td>\n",
       "      <td>47574556</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081894</td>\n",
       "      <td>GSM752590: mdo br M 1</td>\n",
       "      <td>SRR306744</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752590: mdo br M 1</td>\n",
       "      <td>1692798692</td>\n",
       "      <td>22273667</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081895</td>\n",
       "      <td>GSM752592: mdo cb F 1</td>\n",
       "      <td>SRR306745</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752592: mdo cb F 1</td>\n",
       "      <td>1783172952</td>\n",
       "      <td>23462802</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081896</td>\n",
       "      <td>GSM752594: mdo cb M 1</td>\n",
       "      <td>SRR306746</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752594: mdo cb M 1</td>\n",
       "      <td>1789739580</td>\n",
       "      <td>23549205</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081897</td>\n",
       "      <td>GSM752596: mdo ht F 1</td>\n",
       "      <td>SRR306747</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752596: mdo ht F 1</td>\n",
       "      <td>873949460</td>\n",
       "      <td>11499335</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081897</td>\n",
       "      <td>GSM752596: mdo ht F 1</td>\n",
       "      <td>SRR306748</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752596: mdo ht F 1</td>\n",
       "      <td>1348090432</td>\n",
       "      <td>17738032</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081898</td>\n",
       "      <td>GSM752598: mdo ht M 1</td>\n",
       "      <td>SRR306749</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752598: mdo ht M 1</td>\n",
       "      <td>1263916100</td>\n",
       "      <td>16630475</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081898</td>\n",
       "      <td>GSM752598: mdo ht M 1</td>\n",
       "      <td>SRR306750</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752598: mdo ht M 1</td>\n",
       "      <td>2484401620</td>\n",
       "      <td>32689495</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081899</td>\n",
       "      <td>GSM752601: mdo kd F 1</td>\n",
       "      <td>SRR306751</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752601: mdo kd F 1</td>\n",
       "      <td>1653356288</td>\n",
       "      <td>21754688</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081900</td>\n",
       "      <td>GSM752603: mdo kd M 1</td>\n",
       "      <td>SRR306752</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752603: mdo kd M 1</td>\n",
       "      <td>1115625052</td>\n",
       "      <td>14679277</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081901</td>\n",
       "      <td>GSM752606: mdo lv F 1</td>\n",
       "      <td>SRR306753</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752606: mdo lv F 1</td>\n",
       "      <td>1575449752</td>\n",
       "      <td>20729602</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081902</td>\n",
       "      <td>GSM752608: mdo lv M 1</td>\n",
       "      <td>SRR306754</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752608: mdo lv M 1</td>\n",
       "      <td>1444207480</td>\n",
       "      <td>19002730</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081903</td>\n",
       "      <td>GSM752611: mdo ts M 1</td>\n",
       "      <td>SRR306755</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752611: mdo ts M 1</td>\n",
       "      <td>1162273244</td>\n",
       "      <td>15293069</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>SRP007412</td>\n",
       "      <td>SRX081904</td>\n",
       "      <td>GSM752613: mdo ts M 2</td>\n",
       "      <td>SRR306756</td>\n",
       "      <td>13616</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM752613: mdo ts M 2</td>\n",
       "      <td>2598317944</td>\n",
       "      <td>34188394</td>\n",
       "      <td>None</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>149 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    study_accession experiment_accession       experiment_title run_accession  \\\n",
       "0         SRP007412            SRX081869  GSM752557: gga br F 1     SRR306710   \n",
       "1         SRP007412            SRX081870  GSM752558: gga br M 1     SRR306711   \n",
       "2         SRP007412            SRX081871  GSM752559: gga cb F 1     SRR306712   \n",
       "3         SRP007412            SRX081872  GSM752560: gga cb M 1     SRR306713   \n",
       "4         SRP007412            SRX081873  GSM752561: gga ht F 1     SRR306714   \n",
       "5         SRP007412            SRX081874  GSM752562: gga ht M 1     SRR306715   \n",
       "6         SRP007412            SRX081875  GSM752563: gga kd F 1     SRR306716   \n",
       "7         SRP007412            SRX081876  GSM752564: gga kd M 1     SRR306717   \n",
       "8         SRP007412            SRX081877  GSM752565: gga lv F 1     SRR306718   \n",
       "9         SRP007412            SRX081878  GSM752566: gga lv M 1     SRR306719   \n",
       "10        SRP007412            SRX081878  GSM752566: gga lv M 1     SRR306720   \n",
       "11        SRP007412            SRX081879  GSM752567: gga ts M 1     SRR306721   \n",
       "12        SRP007412            SRX081879  GSM752567: gga ts M 1     SRR306722   \n",
       "13        SRP007412            SRX081880  GSM752568: gga ts M 2     SRR306723   \n",
       "14        SRP007412            SRX081881  GSM752569: oan br F 1     SRR306724   \n",
       "15        SRP007412            SRX081881  GSM752569: oan br F 1     SRR306725   \n",
       "16        SRP007412            SRX081882  GSM752570: oan br M 1     SRR306726   \n",
       "17        SRP007412            SRX081882  GSM752570: oan br M 1     SRR306727   \n",
       "18        SRP007412            SRX081883  GSM752571: oan cb F 1     SRR306728   \n",
       "19        SRP007412            SRX081884  GSM752572: oan cb M 1     SRR306729   \n",
       "20        SRP007412            SRX081885  GSM752573: oan ht F 1     SRR306730   \n",
       "21        SRP007412            SRX081886  GSM752575: oan ht M 1     SRR306731   \n",
       "22        SRP007412            SRX081887  GSM752576: oan kd F 1     SRR306732   \n",
       "23        SRP007412            SRX081888  GSM752578: oan kd M 1     SRR306733   \n",
       "24        SRP007412            SRX081888  GSM752578: oan kd M 1     SRR306734   \n",
       "25        SRP007412            SRX081889  GSM752580: oan lv F 1     SRR306735   \n",
       "26        SRP007412            SRX081889  GSM752580: oan lv F 1     SRR306736   \n",
       "27        SRP007412            SRX081890  GSM752581: oan lv M 1     SRR306737   \n",
       "28        SRP007412            SRX081890  GSM752581: oan lv M 1     SRR306738   \n",
       "29        SRP007412            SRX081891  GSM752583: oan ts M 1     SRR306739   \n",
       "..              ...                  ...                    ...           ...   \n",
       "119       SRP007412            SRX081907  GSM752616: mmu br M 2     SRR306762   \n",
       "120       SRP007412            SRX081908  GSM752617: mmu cb F 1     SRR306763   \n",
       "121       SRP007412            SRX081909  GSM752618: mmu cb M 1     SRR306764   \n",
       "122       SRP007412            SRX081910  GSM752619: mmu cb M 2     SRR306765   \n",
       "123       SRP007412            SRX081911  GSM752620: mmu ht F 1     SRR306766   \n",
       "124       SRP007412            SRX081912  GSM752621: mmu ht M 1     SRR306767   \n",
       "125       SRP007412            SRX081913  GSM752622: mmu ht M 2     SRR306768   \n",
       "126       SRP007412            SRX081914  GSM752623: mmu kd F 1     SRR306769   \n",
       "127       SRP007412            SRX081915  GSM752624: mmu kd M 1     SRR306770   \n",
       "128       SRP007412            SRX081916  GSM752625: mmu kd M 2     SRR306771   \n",
       "129       SRP007412            SRX081917  GSM752626: mmu lv F 1     SRR306772   \n",
       "130       SRP007412            SRX081918  GSM752627: mmu lv M 1     SRR306773   \n",
       "131       SRP007412            SRX081919  GSM752628: mmu lv M 2     SRR306774   \n",
       "132       SRP007412            SRX081920  GSM752629: mmu ts M 1     SRR306775   \n",
       "133       SRP007412            SRX081921  GSM752630: mmu ts M 2     SRR306776   \n",
       "134       SRP007412            SRX081893  GSM752588: mdo br F 1     SRR306742   \n",
       "135       SRP007412            SRX081893  GSM752588: mdo br F 1     SRR306743   \n",
       "136       SRP007412            SRX081894  GSM752590: mdo br M 1     SRR306744   \n",
       "137       SRP007412            SRX081895  GSM752592: mdo cb F 1     SRR306745   \n",
       "138       SRP007412            SRX081896  GSM752594: mdo cb M 1     SRR306746   \n",
       "139       SRP007412            SRX081897  GSM752596: mdo ht F 1     SRR306747   \n",
       "140       SRP007412            SRX081897  GSM752596: mdo ht F 1     SRR306748   \n",
       "141       SRP007412            SRX081898  GSM752598: mdo ht M 1     SRR306749   \n",
       "142       SRP007412            SRX081898  GSM752598: mdo ht M 1     SRR306750   \n",
       "143       SRP007412            SRX081899  GSM752601: mdo kd F 1     SRR306751   \n",
       "144       SRP007412            SRX081900  GSM752603: mdo kd M 1     SRR306752   \n",
       "145       SRP007412            SRX081901  GSM752606: mdo lv F 1     SRR306753   \n",
       "146       SRP007412            SRX081902  GSM752608: mdo lv M 1     SRR306754   \n",
       "147       SRP007412            SRX081903  GSM752611: mdo ts M 1     SRR306755   \n",
       "148       SRP007412            SRX081904  GSM752613: mdo ts M 2     SRR306756   \n",
       "\n",
       "     taxon_id library_selection library_layout library_strategy  \\\n",
       "0        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "1        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "2        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "3        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "4        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "5        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "6        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "7        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "8        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "9        9031              cDNA      SINGLE -           RNA-Seq   \n",
       "10       9031              cDNA      SINGLE -           RNA-Seq   \n",
       "11       9031              cDNA      SINGLE -           RNA-Seq   \n",
       "12       9031              cDNA      SINGLE -           RNA-Seq   \n",
       "13       9031              cDNA      SINGLE -           RNA-Seq   \n",
       "14       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "15       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "16       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "17       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "18       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "19       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "20       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "21       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "22       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "23       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "24       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "25       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "26       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "27       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "28       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "29       9258              cDNA      SINGLE -           RNA-Seq   \n",
       "..        ...               ...            ...              ...   \n",
       "119     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "120     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "121     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "122     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "123     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "124     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "125     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "126     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "127     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "128     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "129     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "130     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "131     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "132     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "133     10090              cDNA      SINGLE -           RNA-Seq   \n",
       "134     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "135     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "136     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "137     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "138     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "139     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "140     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "141     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "142     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "143     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "144     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "145     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "146     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "147     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "148     13616              cDNA      SINGLE -           RNA-Seq   \n",
       "\n",
       "     library_source           library_name       bases     spots adapter_spec  \\\n",
       "0    TRANSCRIPTOMIC  GSM752557: gga br F 1  1480245008  19476908         None   \n",
       "1    TRANSCRIPTOMIC  GSM752558: gga br M 1  1334334888  17557038         None   \n",
       "2    TRANSCRIPTOMIC  GSM752559: gga cb F 1  1795697524  23627599         None   \n",
       "3    TRANSCRIPTOMIC  GSM752560: gga cb M 1  1670739920  21983420         None   \n",
       "4    TRANSCRIPTOMIC  GSM752561: gga ht F 1  1748333260  23004385         None   \n",
       "5    TRANSCRIPTOMIC  GSM752562: gga ht M 1  1604929848  21117498         None   \n",
       "6    TRANSCRIPTOMIC  GSM752563: gga kd F 1  1749607628  23021153         None   \n",
       "7    TRANSCRIPTOMIC  GSM752564: gga kd M 1  1732548288  22796688         None   \n",
       "8    TRANSCRIPTOMIC  GSM752565: gga lv F 1  2298690376  30245926         None   \n",
       "9    TRANSCRIPTOMIC  GSM752566: gga lv M 1   653253896   8595446         None   \n",
       "10   TRANSCRIPTOMIC  GSM752566: gga lv M 1  1713238740  22542615         None   \n",
       "11   TRANSCRIPTOMIC  GSM752567: gga ts M 1   666191072   8765672         None   \n",
       "12   TRANSCRIPTOMIC  GSM752567: gga ts M 1  1632872768  21485168         None   \n",
       "13   TRANSCRIPTOMIC  GSM752568: gga ts M 2  2804356756  36899431         None   \n",
       "14   TRANSCRIPTOMIC  GSM752569: oan br F 1   820587580  10797205         None   \n",
       "15   TRANSCRIPTOMIC  GSM752569: oan br F 1  1850093840  24343340         None   \n",
       "16   TRANSCRIPTOMIC  GSM752570: oan br M 1   707293012   9306487         None   \n",
       "17   TRANSCRIPTOMIC  GSM752570: oan br M 1  1037813896  13655446         None   \n",
       "18   TRANSCRIPTOMIC  GSM752571: oan cb F 1  2302429576  30295126         None   \n",
       "19   TRANSCRIPTOMIC  GSM752572: oan cb M 1  2159580280  28415530         None   \n",
       "20   TRANSCRIPTOMIC  GSM752573: oan ht F 1  1429341652  18807127         None   \n",
       "21   TRANSCRIPTOMIC  GSM752575: oan ht M 1  1282580180  16876055         None   \n",
       "22   TRANSCRIPTOMIC  GSM752576: oan kd F 1  1357606316  17863241         None   \n",
       "23   TRANSCRIPTOMIC  GSM752578: oan kd M 1   883208008  11621158         None   \n",
       "24   TRANSCRIPTOMIC  GSM752578: oan kd M 1  1222176520  16081270         None   \n",
       "25   TRANSCRIPTOMIC  GSM752580: oan lv F 1  1205382724  15860299         None   \n",
       "26   TRANSCRIPTOMIC  GSM752580: oan lv F 1  1635037248  21513648         None   \n",
       "27   TRANSCRIPTOMIC  GSM752581: oan lv M 1  1377145688  18120338         None   \n",
       "28   TRANSCRIPTOMIC  GSM752581: oan lv M 1  1501827716  19760891         None   \n",
       "29   TRANSCRIPTOMIC  GSM752583: oan ts M 1  1436304772  18898747         None   \n",
       "..              ...                    ...         ...       ...          ...   \n",
       "119  TRANSCRIPTOMIC  GSM752616: mmu br M 2  1499177976  19726026         None   \n",
       "120  TRANSCRIPTOMIC  GSM752617: mmu cb F 1  3141899660  41340785         None   \n",
       "121  TRANSCRIPTOMIC  GSM752618: mmu cb M 1  1592028164  20947739         None   \n",
       "122  TRANSCRIPTOMIC  GSM752619: mmu cb M 2  2181979000  28710250         None   \n",
       "123  TRANSCRIPTOMIC  GSM752620: mmu ht F 1  3394842784  44668984         None   \n",
       "124  TRANSCRIPTOMIC  GSM752621: mmu ht M 1  1861519756  24493681         None   \n",
       "125  TRANSCRIPTOMIC  GSM752622: mmu ht M 2  1968701036  25903961         None   \n",
       "126  TRANSCRIPTOMIC  GSM752623: mmu kd F 1  3466613232  45613332         None   \n",
       "127  TRANSCRIPTOMIC  GSM752624: mmu kd M 1  1796622064  23639764         None   \n",
       "128  TRANSCRIPTOMIC  GSM752625: mmu kd M 2  2216025784  29158234         None   \n",
       "129  TRANSCRIPTOMIC  GSM752626: mmu lv F 1  3671311252  48306727         None   \n",
       "130  TRANSCRIPTOMIC  GSM752627: mmu lv M 1  1401775616  18444416         None   \n",
       "131  TRANSCRIPTOMIC  GSM752628: mmu lv M 2  2584775808  34010208         None   \n",
       "132  TRANSCRIPTOMIC  GSM752629: mmu ts M 1  1518001808  19973708         None   \n",
       "133  TRANSCRIPTOMIC  GSM752630: mmu ts M 2  2261985720  29762970         None   \n",
       "134  TRANSCRIPTOMIC  GSM752588: mdo br F 1   860751832  11325682         None   \n",
       "135  TRANSCRIPTOMIC  GSM752588: mdo br F 1  3615666256  47574556         None   \n",
       "136  TRANSCRIPTOMIC  GSM752590: mdo br M 1  1692798692  22273667         None   \n",
       "137  TRANSCRIPTOMIC  GSM752592: mdo cb F 1  1783172952  23462802         None   \n",
       "138  TRANSCRIPTOMIC  GSM752594: mdo cb M 1  1789739580  23549205         None   \n",
       "139  TRANSCRIPTOMIC  GSM752596: mdo ht F 1   873949460  11499335         None   \n",
       "140  TRANSCRIPTOMIC  GSM752596: mdo ht F 1  1348090432  17738032         None   \n",
       "141  TRANSCRIPTOMIC  GSM752598: mdo ht M 1  1263916100  16630475         None   \n",
       "142  TRANSCRIPTOMIC  GSM752598: mdo ht M 1  2484401620  32689495         None   \n",
       "143  TRANSCRIPTOMIC  GSM752601: mdo kd F 1  1653356288  21754688         None   \n",
       "144  TRANSCRIPTOMIC  GSM752603: mdo kd M 1  1115625052  14679277         None   \n",
       "145  TRANSCRIPTOMIC  GSM752606: mdo lv F 1  1575449752  20729602         None   \n",
       "146  TRANSCRIPTOMIC  GSM752608: mdo lv M 1  1444207480  19002730         None   \n",
       "147  TRANSCRIPTOMIC  GSM752611: mdo ts M 1  1162273244  15293069         None   \n",
       "148  TRANSCRIPTOMIC  GSM752613: mdo ts M 2  2598317944  34188394         None   \n",
       "\n",
       "     avg_read_length  \n",
       "0               76.0  \n",
       "1               76.0  \n",
       "2               76.0  \n",
       "3               76.0  \n",
       "4               76.0  \n",
       "5               76.0  \n",
       "6               76.0  \n",
       "7               76.0  \n",
       "8               76.0  \n",
       "9               76.0  \n",
       "10              76.0  \n",
       "11              76.0  \n",
       "12              76.0  \n",
       "13              76.0  \n",
       "14              76.0  \n",
       "15              76.0  \n",
       "16              76.0  \n",
       "17              76.0  \n",
       "18              76.0  \n",
       "19              76.0  \n",
       "20              76.0  \n",
       "21              76.0  \n",
       "22              76.0  \n",
       "23              76.0  \n",
       "24              76.0  \n",
       "25              76.0  \n",
       "26              76.0  \n",
       "27              76.0  \n",
       "28              76.0  \n",
       "29              76.0  \n",
       "..               ...  \n",
       "119             76.0  \n",
       "120             76.0  \n",
       "121             76.0  \n",
       "122             76.0  \n",
       "123             76.0  \n",
       "124             76.0  \n",
       "125             76.0  \n",
       "126             76.0  \n",
       "127             76.0  \n",
       "128             76.0  \n",
       "129             76.0  \n",
       "130             76.0  \n",
       "131             76.0  \n",
       "132             76.0  \n",
       "133             76.0  \n",
       "134             76.0  \n",
       "135             76.0  \n",
       "136             76.0  \n",
       "137             76.0  \n",
       "138             76.0  \n",
       "139             76.0  \n",
       "140             76.0  \n",
       "141             76.0  \n",
       "142             76.0  \n",
       "143             76.0  \n",
       "144             76.0  \n",
       "145             76.0  \n",
       "146             76.0  \n",
       "147             76.0  \n",
       "148             76.0  \n",
       "\n",
       "[149 rows x 14 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "copy_sra_data(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1/1 [00:00<00:00, 122.69it/s]\n",
      "100%|██████████| 1/1 [00:00<00:00, 227.98it/s]\n",
      "100%|██████████| 1/1 [00:00<00:00, 234.15it/s]\n",
      "100%|██████████| 5/5 [00:00<00:00, 178.17it/s]\n",
      "100%|██████████| 1/1 [00:00<00:00, 233.61it/s]\n",
      "100%|██████████| 2/2 [00:00<00:00, 250.55it/s]\n",
      "100%|██████████| 14/14 [00:00<00:00, 219.38it/s]\n",
      "100%|██████████| 18/18 [00:00<00:00, 249.19it/s]\n",
      "100%|██████████| 13/13 [00:00<00:00, 222.07it/s]\n",
      "100%|██████████| 10/10 [00:00<00:00, 199.25it/s]\n",
      "100%|██████████| 11/11 [00:00<00:00, 213.72it/s]\n",
      "100%|██████████| 10/10 [00:00<00:00, 187.28it/s]\n",
      "100%|██████████| 8/8 [00:00<00:00, 215.97it/s]\n",
      "100%|██████████| 19/19 [00:00<00:00, 218.56it/s]\n",
      "100%|██████████| 20/20 [00:00<00:00, 250.00it/s]\n",
      "100%|██████████| 15/15 [00:00<00:00, 254.78it/s]\n"
     ]
    }
   ],
   "source": [
    "copy_sra_data(df_paired, sra_dest_dir='/staging/as/skchoudh/rna-seq-datasets/paired')\n",
    "copy_sra_data(df_unpaired, sra_dest_dir='/staging/as/skchoudh/rna-seq-datasets/single')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_config_file(df, samples_to_process_dir, strategy=None):\n",
    "    df_grouped = df.groupby(['taxon_id'])\n",
    "\n",
    "    for taxon_id, df_group in df_grouped:\n",
    "        assert len(\n",
    "            df_group['study_accession'].unique()) == 1, 'Multiple SRPs found'\n",
    "        species = taxon_id_map[taxon_id]\n",
    "        srp = df_group['study_accession'].unique()[0]\n",
    "        filepath = os.path.join(re_ribo_config_dir, '{}_{}.py'.format(species, srp))\n",
    "        if strategy:\n",
    "            filepath = os.path.join(re_ribo_config_dir, '{}_{}_{}.py'.format(species, srp, strategy))\n",
    "            \n",
    "        with open(filepath, 'w') as fh:\n",
    "\n",
    "            config = write_config(species, srp, samples_to_process_dir)\n",
    "            fh.write(config)\n",
    "            print('Wrote {}'.format(filepath))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_config(species, srp, samples_to_process_dir):\n",
    "    rawdata_dir = os.path.join(samples_to_process_dir, species, srp)\n",
    "    out_dir = os.path.join(re_ribo_analysis_dir, species, srp)\n",
    "    cdna_fa = cdna_map[species]    \n",
    "    cdna_idx = cdna_map[species].replace('.fa.gz', '.kallisto.index')\n",
    "    to_write = \"\"\"\n",
    "    RAWDATA_DIR = '{}'\n",
    "    OUT_DIR = '{}'\n",
    "    CDNA_FA_GZ = '{}'    \n",
    "    CDNA_IDX = '{}'    \n",
    "    \"\"\".format(rawdata_dir, out_dir, cdna_fa, cdna_idx)\n",
    "    return dedent(to_write)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/macaca_mulatta_SRP007412_paired.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/gorilla_gorilla_SRP007412_paired.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pan_paniscus_SRP007412_paired.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pan_troglodytes_SRP007412_paired.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pongo_abelii_SRP007412_paired.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/homo_sapiens_SRP007412_paired.py\n"
     ]
    }
   ],
   "source": [
    "create_config_file(df_paired,  '/staging/as/skchoudh/rna-seq-datasets/paired', 'paired')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/gallus_gallus_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/ornithorhynchus_anatinus_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/macaca_mulatta_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/gorilla_gorilla_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pan_paniscus_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pan_troglodytes_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/pongo_abelii_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/homo_sapiens_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/mus_musculus_SRP007412_single.py\n",
      "Wrote /home/cmb-panasas2/skchoudh/github_projects/EE-546-project//snakemake/configs/monodelphis_domestica_SRP007412_single.py\n"
     ]
    }
   ],
   "source": [
    "create_config_file(df_unpaired,'/staging/as/skchoudh/rna-seq-datasets/single', 'single')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:riboraptor]",
   "language": "python",
   "name": "conda-env-riboraptor-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
